lnwallet/btcwallet/btcwallet.go

```diff
@@ -287,15 +287,38 @@ func (b *BtcWallet) InternalWallet() *base.Wallet {
 //
 // This is a part of the WalletController interface.
 func (b *BtcWallet) Start() error {
+	// Is the wallet (according to its database) currently watch-only
+	// already? If it is, we won't need to convert it later.
+	walletIsWatchOnly := b.wallet.Manager.WatchOnly()
+
+	// If the wallet is watch-only, but we don't expect it to be, then we
+	// are in an unexpected state and cannot continue.
+	if walletIsWatchOnly && !b.cfg.WatchOnly {
```

Comment:
Not sure that I understand this. According to the database the wallet is watch-only, yet the config is not? How could that be? Meanwhile shouldn't these two values always be equal?

Reply:
The config here is coming from `--remotesigner.enable`. So if your wallet was created as watch-only but you forgot to turn on `--remotesigner.enable`, then `lnd` will inevitably crash later on because it is missing private keys and can't sign anything. So rather than crashing with a signing error, we return a nice error here. So this is purely to prevent a wrong configuration by the user.